### PR TITLE
LLM Transformers

### PR DESCRIPTION
**Pull request by RonakReyhani**

### CODE REVIEW

File: _posts/2023-07-04-LLM-Transformers.markdown [hunk 1]
```diff
@@ -0,0 +1,264 @@
+---
+layout: postv2
+font: serif
+title: "LLM Transformers"
+description: "Empower LLM models with Transformers and Agents"
```
Comment (Hamo69) [inline _posts/2023-07-04-LLM-Transformers.markdown:5]: LLM Models is Large Language Models Models

File: _posts/2023-07-04-LLM-Transformers.markdown [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+date: 2023-07-04
+author: Nadia Reyhani
+tags: [Machine Learning, OpenAI, StarCoder, OpenAssistant, Transformers, Agents]
+---
+
+## Natural Language Processing and Transformers
+
+<div style="width:100%;height:0;padding-bottom:56%;position:relative;margin-bottom:10%"><iframe src="https://giphy.com/embed/2y1Ns6zIfK6WI8d21y" width="100%" height="100%" style="position:absolute" frameBorder="0" class="giphy-embed" allowFullScreen></iframe></div>
+
+In a recent experiment, I explored Hugging Face and OpenAI Transformers, and in this blog post, I will guide you through some easy steps to generate your first Hugging Face Transformer. But before we dive into the code snippets and their explanations, let's go over some important terms and concepts in Machine Learning.
```
Comment (Hamo69) [inline _posts/2023-07-04-LLM-Transformers.markdown:15]: what is a Hugging Face Transformer??

**Review by Hamo69 — APPROVED** [review]

File: _posts/2023-07-04-LLM-Transformers.markdown (hunk 1 above)
Comment (RonakReyhani) [reply _posts/2023-07-04-LLM-Transformers.markdown:5]: Sorry Hamo, what needs to be fixed here?

File: _posts/2023-07-04-LLM-Transformers.markdown [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+Why do we call them LLM? The term "Large Language Models" is used to emphasize the magnitude and capabilities of these models in handling diverse language-related tasks.
+
+### Transformers
+
+Transformers are a type of architecture used in deep learning models for natural language processing tasks.
+They have revolutionized the field of natural language processing by enabling more accurate and context-aware language understanding and generation. Transformers are designed to understand the relationships between words in a sentence by considering the entire context. This has greatly enhanced the performance of language-related applications, resulting in significant advancements in AI.
+
+### Agents
+
+Transformer Agents are smart software entities that use transformer-based models to interact with users. These agents understand user inputs, context, and generate natural language responses. By leveraging transformers, which excel at understanding word relationships, these agents enable human-like conversations. They are employed in applications like virtual assistants and chatbot, enhancing user experiences and facilitating efficient communication between humans and machines.
```
Comment (carter293) [inline _posts/2023-07-04-LLM-Transformers.markdown:33]: Minor comment: chatbot should be chatbot**_s_**

File: _posts/2023-07-04-LLM-Transformers.markdown [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+
+### Prompt Engineering
+
+Prompt refers to instructions given to language models to elicit desired responses. Prompt Engineering on the other hand, involves carefully crafting input text to guide the model's generation process and shape its output. It is a technique used to enhance the performance and control the output of language models, enabling users to obtain more accurate and tailored responses for their specific needs.
+
+### Hugging Face, Open AI
+
+Hugging Face provides open-source tools including pre-trained models, datasets, and software libraries that make it easier to work with and deploy state-of-the-art NLP models.
+
+OpenAI on the other hand focuses on artificial intelligence (AI) and machine learning. They are known for their advancements in developing large-scale language models like GPT-x
```
Comment (carter293) [inline _posts/2023-07-04-LLM-Transformers.markdown:43]: This GPT-x might be better as just GPT.

**Pull request merged**